File: mvvp-arch/app/src/main/java/thaihn/android/demomvvm/screen/main/MainActivity.java
package thaihn.android.demomvvm.screen.main;

import android.databinding.DataBindingUtil;
import android.os.Bundle;

import java.util.ArrayList;

import thaihn.android.demomvvm.R;
import thaihn.android.demomvvm.data.model.Repository;
import thaihn.android.demomvvm.data.repository.RepoRepository;
import thaihn.android.demomvvm.databinding.ActivityMainBinding;
import thaihn.android.demomvvm.screen.BaseActivity;

public class MainActivity extends BaseActivity {

    private MainViewModel mMainViewModel;
    private MainAdapter mMainAdapter;
    private ActivityMainBinding mBinding;

    @Override
    protected int getLayoutResource() {

        return R.layout.activity_main;
    }

    @Override
    protected void initVariables(Bundle saveInstanceState) {
        DataBindingUtil.setContentView(this, R.layout.activity_main);
    }

    @Override
    protected void initData(Bundle saveInstanceState) {
        mMainAdapter = new MainAdapter(new ArrayList<Repository>());
        RepoRepository repoRepository = RepoRepository.getInstane();
        mMainViewModel = new MainViewModel(mMainAdapter, repoRepository);

        mMainViewModel.getRepo("https://api.github.com/users/google/repos");
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

//        Repository repository = new Repository();
//        repository.setName("Thaihn");
//        repository.setDescription("aaaaaaaa");
//        binding.setRepository(repository);
    }
}

File: clear-architect/presentation/src/main/java/thaihn/com/androidcleanarchitecture/screen/main/MainActivity.kt
package thaihn.com.androidcleanarchitecture.screen.main

import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.lifecycle.ViewModelProviders
import thaihn.com.androidcleanarchitecture.BR
import thaihn.com.androidcleanarchitecture.R
import thaihn.com.androidcleanarchitecture.base.BaseActivity

class MainActivity : BaseActivity() {
  override fun initVariable(savedInstanceState: Bundle?) {
  }

  override fun initData(savedInstanceState: Bundle?) {
    val binding = DataBindingUtil.setContentView<ViewDataBinding>(this, R.layout.activity_main)
    binding.setLifecycleOwner(this)
    binding.setVariable(BR.viewModel, ViewModelProviders.of(this).get(MainViewModel::class.java))
  }

  override fun getLayoutResource(): Int {
    return R.layout.activity_main
  }

}